src/logging.cpp

```diff
@@ -15,36 +15,117 @@
 #include <new>
 #include <string>
 #include <vector>
+#include <mutex>
+#include "environment.h"
 
-#if 0 // Flip this to true for easy local debugging
-static void DefaultLogCallback(sgx_ql_log_level_t level, const char* message)
+using namespace std;
+
+sgx_ql_logging_function_t logger_callback = nullptr;
+static sgx_ql_log_level_t debug_log_level;
+bool enable_debug_log = false;

```

Comment:
>enable_debug_log [](start = 5, length = 16)

Minor suggestion - enable_debug_log is always coupled with debug_log_level. You can define something like "SGX_QL_LOG_NONE = 0" to replace enable_debug_log being false.

Reply:
thanks - fixed